27). Add topic files 79 to 82 (#18081)
dotnet/docs: docs/standard/linq/find-descendant-elements.md

---
title: How to find descendant elements - LINQ to XML

> description: This article shows how to use XPath and LINQ to XML query, in C# and Visual Basic, to find all descendant elements that have a specified name.

Reviewer: ```suggestion
description: Learn how to use XPath and LINQ to XML query, in C# and Visual Basic, to find all descendant elements that have a specified name.
```

ms.date: 07/20/2015
dev_langs:
  - "csharp"
  - "vb"
ms.assetid: b318da39-bb8b-4c56-a019-e13b12b01831
---

# How to find descendant elements (LINQ to XML)

This article shows how to use XPath and LINQ to XML query, in C# and Visual Basic, to find all descendant elements that have a specified name.

> ## Example Find descendant elements that have a specified name

Reviewer: ```suggestion
## Example: Find descendant elements that have a specified name
```

>  This example shows how to use LINQ to XML query and XPath to find all descendant elements named `Name` in XML document [Sample XML file: Multiple purchase orders](sample-xml-file-multiple-purchase-orders.md). The XPath expression is `//Name`.

Reviewer: ```suggestion
This example shows how to use LINQ to XML query and XPath to find all descendant elements named `Name` in XML document [Sample XML file: Multiple purchase orders](sample-xml-file-multiple-purchase-orders.md). The XPath expression is `//Name`.
```

```csharp
XDocument po = XDocument.Load("PurchaseOrders.xml");

// LINQ to XML query
IEnumerable<XElement> list1 = po.Root.Descendants("Name");

// XPath expression
IEnumerable<XElement> list2 = po.XPathSelectElements("//Name");

if (list1.Count() == list2.Count() &&
        list1.Intersect(list2).Count() == list1.Count())
    Console.WriteLine("Results are identical");
else
    Console.WriteLine("Results differ");
foreach (XElement el in list1)
    Console.WriteLine(el);
```

```vb

>       Dim po As XDocument = XDocument.Load("PurchaseOrders.xml")

Reviewer: ```suggestion
Dim po As XDocument = XDocument.Load("PurchaseOrders.xml")
```

' LINQ to XML query
Dim list1 As IEnumerable(Of XElement) = po...<Name>

' XPath expression
Dim list2 As IEnumerable(Of XElement) = po.XPathSelectElements("//Name")

If (list1.Count() = list2.Count() And _
        list1.Intersect(list2).Count() = list1.Count()) Then
    Console.WriteLine("Results are identical")
Else
    Console.WriteLine("Results differ")
End If
For Each el As XElement In list1
    Console.WriteLine(el)
Next
```

> The example produces this output:

Reviewer: ```suggestion
The example produces the following output:
```
